### DISCUSSION TITLE
HTTP API Support for Gravatar default image

### DISCUSSION THREAD

**Discussion by slightlyfaulty** [dicebear/dicebear#223] [Ideas]
Since DiceBear's HTTP API officially supports PNG now 🚀, it would be amazing to be able to use it as a [Gravatar default image](https://en.gravatar.com/site/implement/images/#default-image).

The only roadblock at the moment is the fact that Gravatar doesn't support query strings in the default image URL.

```
MUST NOT include a querystring (if it does, it will be ignored)
```

Of course this means no options can be used in the provided DiceBear URL.

As a workaround, how difficult would it be to add support for passing URL options without a query string? For example something like:

```
https://avatars.dicebear.com/api/initials/size=64&b=red/test.png
```

**Comment by FlorianKoerner** [comment]
Hi @slightlyfaulty 

I've been thinking for a long time about how we could support Gravatar. Your idea to just replace the question mark with a slash is really good! 🥳 

I have already implemented it! But it only works with the new API.
See: https://github.com/dicebear/dicebear/discussions/179

Example:

Before:
https://api.dicebear.com/4.x/adventurer-neutral/png?seed=YOUR-SEED-HERE&backgroundColor=yellow

After:
https://api.dicebear.com/4.x/adventurer-neutral/png/seed=YOUR-SEED-HERE&backgroundColor=yellow

**Comment by slightlyfaulty** [comment]
It works like a charm! 😍

Thank you for implementing this so quickly @FlorianKoerner.

This is a very cool project with some impressive functionality and performance. Keep up the good work!
